lib/Parse/ParseDecl.cpp

```diff
@@ -3639,11 +3638,6 @@ ParserStatus Parser::parseLineDirective(bool isLine) {
       }
       consumeToken(tok::integer_literal);
     }
-
-    LastTokTextEnd = Tok.getText().end();
-    if (parseToken(tok::r_paren, diag::sourceLocation_expected, ")"))
-      return makeParserError();
```

Comment:
Please check `Tok::r_paren` here. We don't want to open virtual file for invalid directive such as `#sourceLocation(file: "test.swift", line: 100-200)`.

Reply:
How about?
```
      if (Tok.isNot(tok::r_paren)) {
        diagnose(Tok, diag::sourceLocation_expected, ")");
        return makeParserError();
      }
d.swift:132:46: error: expected ')' in #sourceLocation directive
#sourceLocation(file: "test.swift", line: 100-200)
                                             ^
```

Reply:
I’ve made an amend with this change.